wikimedia/wikipedia-ios
# Update page related endpoint

**Phabricator:** https://phabricator.wikimedia.org/T365595

### Notes
* This PR updates the calls to use the new Related Pages endpoint
* A new object, RelatedPage, was created based on the new endpoint response.
* This object is now used in the CoreData operations to retrieve article data.
* On this point, I left the older methods that use the summary object since other parts of the app use it. This code is not super DRY, so suggestions are more than welcome

### Test Steps
1. Trigger the "Because you read card" by going through an article
2. Refresh the explorer feed
3. Make sure the content loads. Some articles do not have content for the description
4. Test with Chinese variants; we do not have variant descriptions, so per task description, this is an expected bug (more of a QA note)
